components/org.wso2.config.diff.creater/src/main/java/org/wso2/config/diff/creater/ConfigLoader.java

```diff
@@ -0,0 +1,168 @@
+    private static final Log log = LogFactory.getLog(ConfigLoader.class);
+
+    public void validate(String migratedISHomePath, String defaultISHomePath) throws ConfigMigrateException {
+
+        if (StringUtils.isBlank(migratedISHomePath) && StringUtils.isBlank(defaultISHomePath)) {
+            throw new ConfigMigrateException("Please enter migrated IS home path and default IS home path as inputs.");
+        }
+        if (!new File(migratedISHomePath).exists() && !new File(defaultISHomePath).exists()) {
+            throw new ConfigMigrateException("The given IS home paths are not valid. Please try again!");
+        }
+        log.info("Inputs are successfully validated.");
+    }
+
+    /**
+     * Filter files by the extension and add to maps.
+     *
+     * @param migratedProductHomePath migrated IS home path.
+     * @param defaultProductHomePath default IS home path.
+     * @throws IOException IOException.
+     */
+    public void filterFiles(String migratedProductHomePath, String defaultProductHomePath) throws IOException {
+
+        defaultXMLFiles = getFilesInPath(Paths.get(defaultProductHomePath.concat(MigrationConstants.CONF_PATH)),
+                MigrationConstants.XML_FILE_EXTENSION);
+        migratedXMLFiles = getFilesInPath(Paths.get(migratedProductHomePath.concat(MigrationConstants.CONF_PATH)),
+                MigrationConstants.XML_FILE_EXTENSION);
+
+        defaultPropertiesFiles = getFilesInPath(Paths.get(defaultProductHomePath.concat(MigrationConstants.CONF_PATH)),
+                MigrationConstants.PROPERTY_FILE_EXTENSION);
+        migratedPropertiesFiles = getFilesInPath(Paths.get(migratedProductHomePath.concat(MigrationConstants
+                        .CONF_PATH)), MigrationConstants.PROPERTY_FILE_EXTENSION);
+        j2TemplateFiles = getFilesInPath(Paths.get(defaultProductHomePath.concat(MigrationConstants
+                .TEMPLATE_CONF_PATH)), MigrationConstants.J2_FILE_EXTENSION);
+        log.info("All files are successfully filtered by the extension.");
+    }
+
+    /**
+     * Read the knowledge from hosted  pipe separated CSV file and put that to a Map.
+     *
+     * @param csvRemoteURL Remotely hosted CSV file URL.
+     * @return Map of key and entry.
+     * @throws IOException
+     */
+     public Map<String, String> readFromCSV(URL csvRemoteURL) throws IOException {
+
+        String line;
+        Map<String, String> keys = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(csvRemoteURL.openStream(),
+                StandardCharsets.UTF_8))) {
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(Pattern.quote(MigrationConstants.CSV_SEPARATOR_APPENDER));
+                if (isNewLine(parts)) {
+                    keys.put(parts[TAG_KEY_COLUMN_NO_INDEX].trim(), line);
+                }
+            }
+        }
+        return keys;
+    }
+
+    private boolean isNewLine(String[] parts) {
```

Comment:
For each private method, the name of the method should express its method contract, and the input parameters need to support it. Altogether the name and parameters should explain what it does. Please check other places for the naming of private methods to gain more readability.